sycl/include/CL/sycl/intel/fpga_lsu.hpp

```diff
@@ -0,0 +1,116 @@
+//==-------------- fpga_lsu.hpp --- SYCL FPGA Reg Extensions ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "fpga_utils.hpp"
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/pointers.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace intel {
+constexpr uint8_t BURST_COALESCE = 0x1;
+constexpr uint8_t CACHE = 0x2;
+constexpr uint8_t DONT_STATICALLY_COALESCE = 0x4;
+constexpr uint8_t PREFETCH = 0x8;
+
+template <int32_t N> struct burst_coalesce {
+  static constexpr int32_t value = N;
+  static constexpr int32_t default_value = false;
+};
+
+template <int32_t N> struct cache {
+  static constexpr int32_t value = N;
+  static constexpr int32_t default_value = 0;
+};
+
+template <int32_t N> struct prefetch {
+  static constexpr int32_t value = N;
+  static constexpr int32_t default_value = false;
+};
+
+template <int32_t N> struct dont_statically_coalesce {
```

Comment:
I have the feeling that some of template parameters could be `bool`, even if internally `value` can remain an `int32_t`

Reply:
If you really needs your classes to have always an `int32_t` interface for the implementation, then use a user-facing meta-function, such as
```c++
template <bool B>
using dont_statically_coalesce = dont_statically_coalesce_impl<B>;
```
By the way, would it make sense to rename `dont_statically_coalesce` `statically_coalesce` instead and take the negated boolean?

Reply:
I went with the second option here and I changed `dont_statically_coalesce` to `statically_coalesce`, though this probably requires an update to the SPIR-V spec for consistency (though it works as is for now).

Reply:
@AGindinson to address this comment.

I personally wouldn't like to change SPIR-V spec: as far as I understand by default we expect the coalescence flag to be set in a meaning "do statically coalesce". Therefore with SPIR-V spec change we would result having the appropriate decoration by default in the module (which has the built-in used), while I believe in that 'by default' a module shall have as less decorations as possible.

Reply:
> @AGindinson to address this comment.
> 
> I personally wouldn't like to change SPIR-V spec: as far as I understand by default we expect the coalescence flag to be set in a meaning "do statically coalesce". Therefore with SPIR-V spec change we would result having the appropriate decoration by default in the module (which has the built-in used), while I believe in that 'by default' a module shall have as less decorations as possible.

I agree with your last sentence. Do you have objection to exposing this as `statically_coalesce` instead of `dont_statically_coalesce` in this header though?

Reply:
I'm fine with the naming

Reply:
Yes I think it is good to make the distinction between a simple user-facing UI and the optimized SPIR-V generation. Isn't it the goal of C++? :-)